### ISSUE TITLE
Pressing backspace during quickstart can delete the prompt

### ISSUE THREAD

**Issue by scottbelden** [sphinx-doc/sphinx#5141]
Subject: Pressing backspace during quickstart can delete the prompt

### Problem
- During the `quickstart` setup, if you start typing, then press backspace to delete it all the prompt will disappear

#### Procedure to reproduce the problem
- Begin quickstart:
```
$ sphinx-quickstart 
Welcome to the Sphinx 1.7.5 quickstart utility.

Please enter values for the following settings (just press Enter to
accept a default value, if one is given in brackets).

Selected root path: .

You have two options for placing the build directory for Sphinx output.
Either, you use a directory "_build" within the root path, or you separate
"source" and "build" directories within the root path.
> Separate source and build directories (y/n) [n]: 
```

- Press any key (I chose `a`):
```
$ sphinx-quickstart 
Welcome to the Sphinx 1.7.5 quickstart utility.

Please enter values for the following settings (just press Enter to
accept a default value, if one is given in brackets).

Selected root path: .

You have two options for placing the build directory for Sphinx output.
Either, you use a directory "_build" within the root path, or you separate
"source" and "build" directories within the root path.
> Separate source and build directories (y/n) [n]: a
```

- Press backspace:
```
$ sphinx-quickstart 
Welcome to the Sphinx 1.7.5 quickstart utility.

Please enter values for the following settings (just press Enter to
accept a default value, if one is given in brackets).

Selected root path: .

You have two options for placing the build directory for Sphinx output.
Either, you use a directory "_build" within the root path, or you separate
"source" and "build" directories within the root path.

```

- Notice that the `> Separate source and build directories (y/n) [n]:` prompt is gone

#### Error logs / results
```
N/A
```

#### Expected results
Prompt should stay visible

### Reproducible project / your project
- Any

### Environment info
- OS: Ubuntu 18.04
- Python version: 3.7
- Sphinx version: 175

**Comment by tk0miya** [comment]
This might be related with #5036. So I hope this was already fixed in HEAD of 1.7 branch.
Could you check it please?

**Comment by scottbelden** [comment]
I cloned and tested this on master and the issue is not present. Thanks!
